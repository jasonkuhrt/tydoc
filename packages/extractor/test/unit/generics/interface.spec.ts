describe('first type parameter', () => {
  it('the name', () => {
    const edd = ctx.extract2(/* ts */ `
      export interface Foo<T> {}
    `)
    expect(edd.getInterface('(a).Foo').typeParameters).toMatchSnapshot()
  })

  it('the primitive type default', () => {
    const edd = ctx.extract2(/* ts */ `
      export interface Foo<T = string> {}
    `)
    expect(edd.getInterface('(a).Foo').typeParameters).toMatchSnapshot()
  })

  it('the alias object type default is added to the type index and refererenced', () => {
    const edd = ctx.extract2(/* ts */ `
      export type Bar = { a: 1 }
      export interface Foo<T = Bar> {}
    `)
    expect(edd.getInterface('(a).Foo').typeParameters).toMatchSnapshot()
    expect(edd.getAlias('(a).Bar')).toMatchSnapshot()
  })

  it('the interface object type default is added to the type index and refererenced', () => {
    const edd = ctx.extract2(/* ts */ `
      export interface Bar { a: 1 }
      export interface Foo<T = Bar> {}
    `)
    expect(edd.getInterface('(a).Foo').typeParameters).toMatchSnapshot()
    expect(edd.getInterface('(a).Bar')).toMatchSnapshot()
  })
})
