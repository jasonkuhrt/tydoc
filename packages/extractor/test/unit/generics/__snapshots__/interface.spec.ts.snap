// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first type parameter the alias object type default is added to the type index and refererenced 1`] = `
Array [
  Object {
    "default": Object {
      "kind": "typeIndexRef",
      "link": "(a).Bar",
    },
    "name": "T",
    "raw": Object {
      "nodeFullText": "T = Bar",
      "nodeText": "T = Bar",
      "typeText": "T",
    },
  },
]
`;

exports[`first type parameter the alias object type default is added to the type index and refererenced 2`] = `
Object {
  "kind": "alias",
  "name": "Bar",
  "raw": Object {
    "nodeFullText": "export type Bar = { a: 1 };",
    "nodeText": "export type Bar = { a: 1 };",
    "typeText": "Bar",
  },
  "tsdoc": null,
  "type": Object {
    "kind": "object",
    "props": Array [
      Object {
        "kind": "prop",
        "name": "a",
        "type": Object {
          "base": "number",
          "kind": "literal",
          "name": "1",
        },
      },
    ],
    "raw": Object {
      "nodeFullText": "export type Bar = { a: 1 };",
      "nodeText": "export type Bar = { a: 1 };",
      "typeText": "Bar",
    },
  },
  "typeParameters": Array [],
}
`;

exports[`first type parameter the interface object type default is added to the type index and refererenced 1`] = `
Array [
  Object {
    "default": Object {
      "kind": "typeIndexRef",
      "link": "(a).Bar",
    },
    "name": "T",
    "raw": Object {
      "nodeFullText": "T = Bar",
      "nodeText": "T = Bar",
      "typeText": "T",
    },
  },
]
`;

exports[`first type parameter the interface object type default is added to the type index and refererenced 2`] = `
Object {
  "kind": "interface",
  "name": "Bar",
  "props": Array [
    Object {
      "kind": "prop",
      "name": "a",
      "type": Object {
        "base": "number",
        "kind": "literal",
        "name": "1",
      },
    },
  ],
  "raw": Object {
    "nodeFullText": "export interface Bar {
  a: 1;
}",
    "nodeText": "export interface Bar {
  a: 1;
}",
    "typeText": "Bar",
  },
  "tsdoc": null,
  "typeParameters": Array [],
}
`;

exports[`first type parameter the name 1`] = `
Array [
  Object {
    "default": null,
    "name": "T",
    "raw": Object {
      "nodeFullText": "T",
      "nodeText": "T",
      "typeText": "T",
    },
  },
]
`;

exports[`first type parameter the primitive type default 1`] = `
Array [
  Object {
    "default": Object {
      "kind": "primitive",
      "type": "string",
    },
    "name": "T",
    "raw": Object {
      "nodeFullText": "T = string",
      "nodeText": "T = string",
      "typeText": "T",
    },
  },
]
`;
